odb-convert/src/main/scala/flatgraph/convert/Convert.scala

```diff
@@ -13,27 +15,60 @@ import java.util.concurrent.atomic.AtomicLong
 import scala.collection.mutable
 
 object Convert {
+  val ANON_EDGE_PROPERTY = "EDGE_PROPERTY"
 
   def main(args: Array[String]): Unit = {
-    if (args.length < 2) {
-      System.err.println("Usage: convert [inputfile] [outputfile]")
-      System.err.println("Error: missing input and/or output file - exiting.")
+    if (args.length == 2) {
+      convertOdbToFlatgraph(overflowDbFile = Paths.get(args(0)), outputFile = Paths.get(args(1)))
+    } else if (args.length == 3 && args(0) == "-r") {
+      convertFlatgraphToOdb(fgFile = Paths.get(args(1)), outputFile = Paths.get(args(2)))
     } else {
-      apply(overflowDbFile = Paths.get(args(0)), outputFile = Paths.get(args(1)))
+      System.err.println("Usage: convert [inputfileOdb] [outputfileFlatGraph]")
+      System.err.println("Usage: convert -r [inputfileFlatGraph] [outputfileOdb]")
+      System.err.println(s"""Seen: ${args.mkString(",")}""")
+      System.err.println("Error: missing input and/or output file - exiting.")
     }
   }
 
-  def apply(overflowDbFile: Path, outputFile: Path): Unit = {
+  def convertOdbToFlatgraph(overflowDbFile: Path, outputFile: Path): Unit = {
     val storage = overflowdb.storage.OdbStorage.createWithSpecificLocation(overflowDbFile.toFile, new overflowdb.util.StringInterner)
     val (nodes, strings) = readOdb(storage)
     writeData(outputFile.toFile, nodes, strings)
   }
 
-  private class NodeRefTmp(val legacyId: Long) {
+  def convertFlatgraphToOdb(fgFile: Path, outputFile: Path, debug: Boolean = false): Unit = {
+    val fg = flatgraph.storage.Deserialization.readGraph(fgFile, None, false)
+
+    val dst = outputFile.toFile
+    if (dst.exists()) dst.delete()
+    dst.createNewFile()
+    val storage = overflowdb.storage.OdbStorage.createWithSpecificLocation(dst, new overflowdb.util.StringInterner)
+    for (node <- fg.allNodes if !AccessHelpers.isDeleted(node)) {
+      val bytes = packNode(node, storage)
+      if (debug) {
+        val legacyIdToNewId = mutable.HashMap[Long, NodeRefTmp]()
+        val stringInterner  = mutable.LinkedHashMap[String, StringRef]()
+        val byLabel         = mutable.LinkedHashMap[String, NodeStuff]()
+        try {
+          readNode(legacyIdToNewId, stringInterner, node.id(), bytes, byLabel, storage)
+        } catch {
+          case exc: Throwable =>
+            println(s"fuck ${node.seq()} / ${node.nodeKind} / ${node.label()} / ${node.id()}")
```

Comment:
i guess this is part of your "debug output" that will be removed / censored before merge? :)